components/apimgt/org.wso2.carbon.apimgt.micro.gateway/org.wso2.carbon.apimgt.micro.gateway.common/src/main/java/org/wso2/carbon/apimgt/micro/gateway/common/api/APIExecutionHandler.java

```diff
@@ -0,0 +1,104 @@
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.governance.api.generic.GenericArtifactManager;
+import org.wso2.carbon.governance.api.generic.dataobjects.GenericArtifact;
+
+import org.wso2.carbon.registry.core.Resource;
+import org.wso2.carbon.registry.core.exceptions.RegistryException;
+import org.wso2.carbon.registry.core.jdbc.handlers.RequestContext;
+
+import java.util.Map;
+
+/**
+ * This class consists methods that will create, prototype, publish, block, deprecate and
+ * retire  an API to API Manager.
+ * <p/>
+ * This executor used to publish a service to API store as a API.
+ *
+ */
+public class APIExecutionHandler extends APIExecutor {
+
+    Log log = LogFactory.getLog(APIExecutor.class);
+
+    /**
+     * This method is called when the execution class is initialized.
+     * All the execution classes are initialized only once.
+     *
+     * @param parameterMap Static parameter map given by the user.
+     *                     These are the parameters that have been given in the
+     *                     lifecycle configuration as the parameters of the executor.
+     */
+    public void init(Map parameterMap) {
+    }
+
+    /**
+     * @param context      The request context that was generated from the registry core.
+     *                     The request context contains the resource, resource path and other
+     *                     variables generated during the initial call.
+     * @param currentState The current lifecycle state.
+     * @param targetState  The target lifecycle state.
+     * @return Returns whether the execution was successful or not.
+     */
+    public boolean execute(RequestContext context, String currentState, String targetState) {
+        APIExecutor apiExecutor = new APIExecutor();
+        try {
+            if (log.isDebugEnabled()) {
+                log.debug("Executing custom API executor for capturing API publish/re-publish events.");
+            }
+
+            String domain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            GenericArtifactManager artifactManager = APIUtil
+                    .getArtifactManager(context.getSystemRegistry(), APIConstants.API_KEY);
+            String artifactId = context.getResource().getUUID();
+            GenericArtifact apiArtifact = artifactManager.getGenericArtifact(artifactId);
+            API api = APIUtil.getAPI(apiArtifact);
+
+            // Add API publish/Re-publish events to database
+            APIIdentifier apiId = api.getId();
+            OnPremiseGatewayDAO onPremiseGatewayDao = new OnPremiseGatewayDAO();
+            onPremiseGatewayDao.addAPIPublishEvent(domain, apiId.toString());
+
+            if (log.isDebugEnabled()) {
+                log.debug("Successfully captured API publish/re-publish event.");
+            }
+        } catch (RegistryException e) {
+            log.error("Failed to get the generic artifact while executing APIExecutor. ", e);
```

Comment:
Whats the reason for allowing to execute the logic in the default executor(APIExecutor) only logging the error.
Whats the purpose of not returning false here.

Reply:
This is because if an exception were to arise here, we still need to prevent it from affecting the default functionality. Hence the exception is logged and not thrown.